camera-manager.js

```diff
@@ -121,29 +121,96 @@ const cameraManager = {
       return value1 + (value2 - value1) * amount;
     }
 
+    function checkRayIntersection(origin,dirQuaternion,maxDistance)
+    {
+      let res = false;
+
+      let collision = physicsManager.raycast(origin, dirQuaternion);
```

Comment:
We could actually make the API support taking multiple raycasts at a time.

Reply:
That should help with performance. Since this is run every frame, it might be worth it.

Reply:
Ok, I am working on API for multiple raycasts at a time.
And that will change a lot of this code.